imports/plugins/core/orders/server/no-meteor/mutations/placeOrder.js

```diff
@@ -170,8 +171,25 @@ export default async function placeOrder(context, input) {
     fulfillmentGroups,
     shopId
   } = orderInput;
+  let { language } = orderInput;
   const { accountId, account, collections, getFunctionsOfType, userId } = context;
-  const { Orders } = collections;
+  const { Orders, Shops } = collections;
+
+  // first search for shop based on account id
+  let shop = await Shops.findOne({ _id: shopId }, { languages: 1 });
+
+  if (!shop || !shop.languages || shop.languages.length === 0) {
+    // if non-primary shop does not have any languages use primary shop
+    shop = await Shops.findOne({ shopType: "primary" }, { languages: 1 });
+  }
+
+  const shopLanguages = (shop && shop.languages) || [];
+  const shopLanguage = R.find(R.whereEq({ enabled: true, i18n: language }))(shopLanguages);
+
+  // set to undefined value so order language will not be set
+  if (!shopLanguage) {
+    language = undefined;
```

Comment:
Throwing a helpful error would be better so they aren't confused why the language they specified isn't set.